File: 189_RotateArray/Solution.py
class Solution:
    # @param {integer[]} nums
    # @param {integer} k
    # @return {void} Do not return anything, modify nums in-place instead.
    def rotate(self, nums, k):
        k = len(nums) - (k % len(nums)) -1
        t = nums[:k+1]
        for i in range(k+1, len(nums)):
            nums[i-k-1] = nums[i]
        for i in range(len(nums)-k-1, len(nums)):
            nums[i] = t[i-len(nums)+k+1]

s = Solution()
xs = range(1, 2)
xs = range(1, 8)
xs = [-1]
s.rotate(xs, 2)
print xs

File: 189_RotateArray/Solution.java
class Solution {
  public void rotate(int[] nums, int k) {
    k = k % nums.length;
    int rest[] = new int[k];
    for (int i = 0; i < k; i++) {
      rest[i] = nums[nums.length-k+i];
    }
    for (int i = nums.length-k-1; i >= 0; i--) {
      nums[k+i] = nums[i];
    }
    for (int i = 0; i < k; i++) {
      nums[i] = rest[i];
    }
  }
}
